database/database_virtual_accounts.go

```diff
@@ -188,30 +189,13 @@ func GetMaxVANum(accNum string, db *sql.DB) (maxId int, err error) {
 }
 
 func RevertVacBalanceToMainAccount(db *sql.DB, va models.VirtualAccounts) (err error) {
-	acc, err := GetAccountByAccountNum(db, va.AccountNum)
-
 	if err == nil {
-		_, err = db.Exec("UPDATE accounts SET account_balance=$1 WHERE account_id=$2;", acc.AccountBalance+va.VaBalance, acc.AccountId)
+		_, err = db.Exec(fmt.Sprintf("UPDATE accounts SET account_balance = account_balance + subquery.va_balance FROM (SELECT va_balance FROM virtual_accounts WHERE va_num = '%s') as subquery WHERE account_num = '%s'; DELETE FROM virtual_accounts WHERE va_num = '%s';", va.VaNum, va.AccountNum, va.VaNum))
 	}
 	return
 }
 
-func DeleteVacById(db *sql.DB, vId int) (err error) {
-	_, err = db.Exec("DELETE FROM virtual_accounts WHERE va_id=$1;", vId)
-	return
-}
-
-func GetAccountByAccountNum(db *sql.DB, accountNum string) (acc models.Accounts, err error) {
-	err = db.QueryRow("SELECT account_id, account_num, account_balance FROM accounts WHERE account_num=$1", accountNum).Scan(&acc.AccountId, &acc.AccountNum, &acc.AccountBalance)
-	return
-}
-
-func GetCustomerById(db *sql.DB, id int) (cust models.Customers, err error) {
-	err = db.QueryRow("SELECT cust_id, account_num, cust_email FROM customers WHERE cust_id=$1;", id).Scan(&cust.CustId, &cust.AccountNum, &cust.CustEmail)
-	return
-}
-
 func GetVacByAccountNum(db *sql.DB, accountNum string) (va models.VirtualAccounts, err error) {
-	err = db.QueryRow("SELECT va_id, va_num, account_num, va_balance FROM virtual_accounts WHERE account_num=$1", accountNum).Scan(&va.VaId, &va.VaNum, &va.AccountNum, &va.VaBalance)
+	err = db.QueryRow("SELECT va_id, va_num, account_num, va_balance FROM virtual_accounts WHERE account_num=$1 FOR UPDATE;", accountNum).Scan(&va.VaId, &va.VaNum, &va.AccountNum, &va.VaBalance)
```

Comment:
afaik, SELECT...FOR UPDATE can only be useful when done inside a transaction. So you can either pass a transaction (result of db.Begin()) as a parameter or you can run the query under one transaction inside the function body